src/addr.rs

```diff
@@ -567,6 +582,36 @@ impl MaxLenPrefix {
     }
 }
 
+// The ordering of MaxLenPrefixes is similar to the ordering of Prefixes. The
+// only difference is the optional MaxLen. When two prefixes are equal but
+// differ in (presence of) max_len, the order is as follows:
+// - any max_len always comes before no max_len
```

Comment:
I think it would be better if a missing max length would sort right before an existing max length equal to the prefix length – because a missing max length basically means a max length equal to the prefix length.

Reply:
I've added some clarification on this in the comments: it does not matter in practice as the max_len can never be lower than the actual prefix length.